api/etcdserverpb/raft_internal.proto

```diff
@@ -38,11 +43,11 @@ message InternalRaftRequest {
 
   AlarmRequest alarm = 10;
 
-  LeaseCheckpointRequest lease_checkpoint = 11;
+  LeaseCheckpointRequest lease_checkpoint = 11 [(versionpb.etcd_version_field) = "3.4"];
 
   AuthEnableRequest auth_enable = 1000;
   AuthDisableRequest auth_disable = 1011;
-  AuthStatusRequest auth_status = 1013;
+  AuthStatusRequest auth_status = 1013 [(versionpb.etcd_version_field) = "3.5"];
```

Comment:
Did you have to manually write these versions, if so will they change in future release will we have to keep bumping them?

Reply:
No need to bump them, those annotations indicate etcd version that introduced the specific field/message. However any new message/field added after this change should include those annotations.

Reply:
Thanks for that.

> However any new message/field added after this change should include those annotations.

is this enforced in any way? e.g. with tests or failure of proto generation?

Reply:
+1 to enforcing. 
If we had no 'message level annotations', it would be easier to enforce no fields without an annotation
in the transitive proto-descriptor.

Reply:
Enforcing is definitely in the plan. It might be a little harder to implement with message and enum annotations, but it should be possible if we introduce allowlisting (enforcer will require always message version set, but for fields only if they are not in allow list maintained by enforcer).

Decided on splitting message vs field and enum vs enum value based on:
* Enum default value (equal 0) will never be considered set, so annotating it doesn't make sense. It can be also confusing as someone might think that setting this enum value will mean that required etcd version is changed, but it's not.
* Messages with no fields, for example AuthStatusRequest. We technically can determine the version based on field annotation that uses this message, but I think it might become confusing when we will add some fields to this message. Decided that messages should be self sufficient to define their etcd version.
* Readability, having this annotation on each and every field makes the proto much less readable. I would prefer to keep the proto definition cleaner and even if the enforces will be more complicated.

Reply:
Enforcing of the version is not done to all the proto files, like lease.proto doesn't have it, so how do we determine going forward which require a version field in a tested way?

Reply:
For storage versioning we just need to ensure that protos used in WAL have annotations. This means that we should look for all submessages from `raftpb.Entry` as a root. We might consider expanding this to all proto if that would useful for other purposes.